### DISCUSSION TITLE
Issue connecting Classic robot cables to V3 DQLA1-based controllers

### DISCUSSION THREAD

**Discussion by yafei-ou** [jhu-dvrk/sawIntuitiveResearchKit#202] [Q&A] [answered]
I am trying to test two newly arrived V3 DQLA1 PSM controllers. However, I cannot fully connect and lock the ITT Cannon cable (the lock can not be fully rotated to the 90 degree lock position although I have tried with a large torque, and thus the connection cannot be established). I wonder what might have caused the issue or if I missed something. This didn't happen to the previous version of controllers, and I have tried two PSM robots on both new controllers, but none of the attempts was successful.

Thanks so much for the help.

![image_123650291 (2)](https://github.com/jhu-dvrk/sawIntuitiveResearchKit/assets/91760846/0e833065-4781-485f-acf8-60a3024d0336)
![image_6209779](https://github.com/jhu-dvrk/sawIntuitiveResearchKit/assets/91760846/ad03e3d7-b7d2-4cac-9bdf-1d9b27b10847)

**Comment by pkazanzides** [comment]
Is it possible that the cutout on the enclosure is not large enough and the connector is not fully seated?

**Reply by yafei-ou** [reply]
Thanks for your reply. I am not sure what you mean by "cutout on the enclosure" (do you mean the cutout of the opening part on the box surface?), but I also have the guess that it might not be fully seated. It seems that the cable fits in the port on the controller, but could it be that the port is placed too deep from the surface of the box and there is still still a distance between both sides due to this stud-like object?

![image_123650291](https://github.com/jhu-dvrk/sawIntuitiveResearchKit/assets/91760846/bded0c3c-dd05-45c5-a2d8-cbc386ddc4bf)

**Comment by adeguet1** [comment]
Here is a view of our MTM DQLA controller.

---

On Apr 17, 2024, at 1:40 PM, Yafei Ou ***@***.***> wrote:

﻿

Thanks for your reply. I am not sure what you mean by "cutout on the enclosure" (do you mean the cutout of the opening part on the box surface?), but I also have the guess that it might not be fully seated. It seems that the cable fits in the port on the controller, but could it be that the port is placed too deep from the surface of the box and there is still still a distance between both sides due to this stud-like object?

image_123650291.JPG (view on web)<https://github.com/jhu-dvrk/sawIntuitiveResearchKit/assets/91760846/bded0c3c-dd05-45c5-a2d8-cbc386ddc4bf>

**Comment by adeguet1** [comment]
I sent too early.   I also check our older PSM controllers and it seems that:

- the “stud like” object is attached to the PSM connector itself

- the dMIB was not placed as deep inside the enclosure

Is it easy to remove the short stubs from the PSM cable?   This might be the simplest solution.

Anton


[image0.jpeg]But
---

On Apr 17, 2024, at 2:15 PM, Anton Deguet ***@***.***> wrote:

﻿ Here is a view of our MTM DQLA controller.

---

On Apr 17, 2024, at 1:40 PM, Yafei Ou ***@***.***> wrote:

﻿

Thanks for your reply. I am not sure what you mean by "cutout on the enclosure" (do you mean the cutout of the opening part on the box surface?), but I also have the guess that it might not be fully seated. It seems that the cable fits in the port on the controller, but could it be that the port is placed too deep from the surface of the box and there is still still a distance between both sides due to this stud-like object?

image_123650291.JPG (view on web)<https://github.com/jhu-dvrk/sawIntuitiveResearchKit/assets/91760846/bded0c3c-dd05-45c5-a2d8-cbc386ddc4bf>

**Reply by yafei-ou** [reply]
Thank you. Removing the short stubs from the PSM cable seems to need disassembling the connector. I am not sure how risky that can be, and I'm not sure if it is useful for some parts inside the connector or not. Is there anything that can be done from the controller side?

PS: There is still not any images attached to your message.

**Reply by adeguet1** [reply]
![image](https://github.com/jhu-dvrk/sawIntuitiveResearchKit/assets/1120243/5183f56d-0f51-4784-9c32-42a08aeb943b)

**Reply by keshuaixu** [reply]
The “studs” are for keying purpose and they are not really needed for dVRK. They can be removed by removing a screw from inside.

https://mm.digikey.com/Volume0/opasdata/d220001/medias/docus/2185/DL_Series_ZIF_Con_Cat.pdf

see page 37

Can you take a picture of the connector on controller without the cable from an oblique angle? Trying to see if the connector is too deeply recessed or the cutout on the sheet metal is not big enough.

**Reply by yafei-ou** [reply]
Thanks for the information. 

> Can you take a picture of the connector on controller without the cable from an oblique angle? Trying to see if the connector is too deeply recessed or the cutout on the sheet metal is not big enough.

Below are two images. To me the connector doesn't seem to be very deep and the cutout on the metal seems to be fitting.

![image_6209779 (1)](https://github.com/jhu-dvrk/sawIntuitiveResearchKit/assets/91760846/a7622b4c-643b-4ce2-89ad-ec8cbb0a60a3)
![image_6209779 (3)](https://github.com/jhu-dvrk/sawIntuitiveResearchKit/assets/91760846/18dcd35d-d85e-4a4a-8791-7537fe3e69c3)

I also noticed that the stubs only exist for PSM and ECM cables, not the MTM ones. They are also very close (directly contacting) the metal surface of the controller boxes in the old version, so I am really not sure if that is the reason.

**Comment by yafei-ou** [comment]
Thank you all for your help. I can now lock the connector after removing the studs. It seems that the new controllers have a slightly deeper-placed connector.

There are still some issues remaining. After powering on the controller, the LED lights on the front side of the controller does not light up for sections "Card A" and "Card B". Is this a desired behavior (the old controllers will light up some LEDs if I am not mistaken)?

For now, I cannot power on the robot. The following errors are raised by the console app. I'm not sure if this is related to the LED lights being not on, or a robot connection issue.

> -W Class mtsRobotIO1394: CheckState: PSM1, actuator 0 power: -6.2729 > limit: 0.2
> -W Class mtsRobotIO1394: CheckState: PSM1, actuator 1 power: -6.24304 > limit: 0.2
> -W Class mtsRobotIO1394: CheckState: PSM1, actuator 2 power: 6.26194 > limit: 0.2
> -W Class mtsRobotIO1394: CheckState: PSM1, actuator 3 power: -6.24982 > limit: 0.2
> -W Class mtsRobotIO1394: CheckState: PSM1, actuator 4 power: -6.24982 > limit: 0.2
> -W Class mtsRobotIO1394: CheckState: PSM1, actuator 5 power: 6.25486 > limit: 0.2
> -W Class mtsRobotIO1394: CheckState: PSM1, actuator 6 power: 6.24982 > limit: 0.2
> E- cmnThrow with std::exception (PSM1: too many consecutive current safety violations.  Power has been disabled.)
> -E Class mtsRobotIO1394: File: mtsRobotIO1394.cpp Line: 452 -  PostRead: PSM1: standard exception "PSM1: too many consecutive current safety violations.  Power has been disabled."
> -E Class mtsRobotIO1394: io:PSM1: 1.7134e+09s, #15335: IO exception: PSM1, PSM1: too many consecutive current safety violations.  Power has been disabled.
> -E Class mtsIntuitiveResearchKitConsole: console:Main: 1.7134e+09s, #15335: IO exception: PSM1, PSM1: too many consecutive current safety violations.  Power has been disabled.

Generally, is there a documentation for connecting the controllers (physically)?

**Comment by pkazanzides** [comment]
Is there a MicroSD card in the slot?  The system needs to load the correct firmware from that card; otherwise, there will just be a boot loader.  Can you run the "qladisp" command on the terminal?  It will list all boards that it finds on the bus, as well as details about the hardware/firmware versions.  For the DQLA controller, you should see DQLA.  The boot firmware shows up as BCFG.

**Reply by yafei-ou** [reply]
There is one MicroSD card in the slot inserted already when we unbox it. Below is the qladisp results:

> Trying to detect boards on port:
> FirewirePort::Init: number of ports = 1
>   Port 0: /dev/fw1, 2 nodes
> FirewirePort::Init: successfully initialized port 0
> Using libraw1394 version 2.1.2
> FirewirePort::Init: successfully disabled cycle start packet
> FirewirePort::InitNodes: base node id = ffc0
> BasePort::ScanNodes: building node map for 1 nodes:
>   Node 0, BoardId = 8, FPGA_V3, Hardware = DQLA, Firmware Version = 8
> BasePort::ScanNodes: found 1 boards
> BasePort::SetDefaultProtocol: all nodes broadcast capable and support shorter wait

**Reply by pkazanzides** [reply]
That all looks good.

**Comment by pkazanzides** [comment]
You can run "qladisp 8" (i.e., qladisp followed by the board number) to see if the readings look o.k.  For example, the encoder and pot positions should change when you move the PSM joints.  Perhaps a screenshot would help.

**Reply by yafei-ou** [reply]
Thanks. I deleted my previous reply. Here is the screenshot when running `qladisp 8`:

![image](https://github.com/jhu-dvrk/sawIntuitiveResearchKit/assets/91760846/2c1b0b60-ab0e-469f-937c-72b44c35bb3e)

**Reply by yafei-ou** [reply]
I re-run the command, and it gives this now:
![image](https://github.com/jhu-dvrk/sawIntuitiveResearchKit/assets/91760846/3420f553-3f99-42ed-913b-026b248c63bb)

**Reply by pkazanzides** [reply]
Thanks.  The display will look better if you increase the size of the window.  It seems like there may be some Firewire communication problems.

**Reply by yafei-ou** [reply]
Sorry. Here it is:
![image](https://github.com/jhu-dvrk/sawIntuitiveResearchKit/assets/91760846/f882b1b5-1567-4672-88c0-4dde19b92bbb)
From my understanding, the read values seem all invalid, is that right?

**Reply by pkazanzides** [reply]
There seems to be a problem with the analog feedback.  It is reading FFFF for both the potentiometer and motor current.  Have you tried the other controller box?

**Reply by pkazanzides** [reply]
At least the Pot and Cur (motor current) are invalid.  The others look o.k., but the real proof would be if Enc and Vel change when you manually move the arm.

**Reply by yafei-ou** [reply]
I have tried the other controller (boardID 6), which seems to give the same results. I have also tried to see if I messed up with the robot connector when removing the studs, but the robot is still working on the old controllers with software version 2.1.

![image](https://github.com/jhu-dvrk/sawIntuitiveResearchKit/assets/91760846/dc63c2b0-597b-481b-8b17-9ce592f44f7c)

**Reply by pkazanzides** [reply]
I have never seen this problem before.  Perhaps there is a cabling problem inside the controller.  I would especially check the two 50-pin FFC (flat-flex cables) that go between the boards.  Perhaps they came loose during shipping.

**Reply by yafei-ou** [reply]
Thank you, I will open the box tomorrow and check. Just in case, did I do anything wrong when connecting the cables? I am only connecting one PSM, one PC, and one controller at a time:

![image_123650291](https://github.com/jhu-dvrk/sawIntuitiveResearchKit/assets/91760846/47d77828-6192-43a7-8d5d-beb21da19b85)

**Reply by pkazanzides** [reply]
Your connection looks fine to me.

**Comment by yafei-ou** [comment]
It turns out that in both controllers, the QLA-Q boards are disconnected. I have reconnected one of them, but for the other one, one pin on QLA1 is bent possibly when the board went loose. What should I do with this? Should I try to bend it back myself? Please see image below. The second pin in the right row is bent.

![image_241090803](https://github.com/jhu-dvrk/sawIntuitiveResearchKit/assets/91760846/4a011ce6-98d3-4774-9600-c56ada7ed8e2)

**Comment by adeguet1** [comment]
For reference, we have a description for all the dVRK controllers on the Wiki.  For the QLA1 and DQLA based, see https://github.com/jhu-dvrk/sawIntuitiveResearchKit/wiki/Controller-Boxes#internal-components

Regarding the bent pin, you should be able to straighten it.   For the other controller, the one for which you re-seated the DQLA-Q board, I would recommend we schedule a Zoom meeting so we can first check that all the signals make sense.  A PID tracking error could be caused by a bad encoder or no power to a motor.  You can email me @jhu.edu to schedule the Zoom meeting.

**Comment by yafei-ou** [comment, marked as answer]
Thank you everyone for your help. I will close this discussion as both the controllers seem to be working now. If I have further issues related to this, I will re-open it. For the other issues related to the software/encoder/potentiometer (PID tracking), I will have meetings with Anton and open another discussion if needed.

Just for later reference (for other people), here is the information related to solving the issues:

- It seems that the new V3 controller has placed the Robot Connector deeper than previous versions. I removed the two "studs" in the PSM cable connector.
- The DQLA-Q board that connects two QLAs went loose during shipping. I guess that is also why the front LEDs in sections "Card A" and "Card B" on the controller never light up.
